### PR TITLE
Homework5: Yuliya Golovach

### PR DESCRIPTION
**Pull request by YuliyaGolovach**

### CODE REVIEW

File: src/hw_5/functions_to_run/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from .digits_multiplication import test_digits_multiplication
+from .fibonacci import solve_fibonacci_task
+from .palindrome import is_palindrome
+from .the_longest_word import solve_the_longest_word_task
+
+
+__all__ = ['test_digits_multiplication',
```
Comment (belovmd) [inline src/hw_5/functions_to_run/__init__.py:7]: great

File: src/hw_5/t02_decorator.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+RANDOM_RANGE = 100
+
+
+def set_call_result(f_name, f_result):
+    lines = []
+    function_found = False
+    if path.exists(FILE_NAME):
+        with open(FILE_NAME, 'r') as f:
+            for line in f:
+                function_name, res = line.split(DICT_SEP)
```
Comment (belovmd) [inline src/hw_5/t02_decorator.py:25]: you can use pickle module here (serialization/deserialization). But it's great. we can also add datetime.

File: src/hw_5/t06_find_max_divisor.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+""" Find max divisor
+
+Find a maximum divisor d of a number n, d = 2**k.
+Example:
+    10(2) 16(16), 12(4).
+"""
+
+
+def get_max_divisor(n):
```
Comment (belovmd) [inline src/hw_5/t06_find_max_divisor.py:9]: amazing. 
I don't understand how have you found this solution. 
My solution was with shifts till last wouldn't be 1.

**Review by belovmd — APPROVED** [review]

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-python-spring/blob/master/homework_review/Homework4.md):
-

File: src/hw_5/t06_find_max_divisor.py (hunk 3 above)
Comment (YuliyaGolovach) [reply src/hw_5/t06_find_max_divisor.py:9]: Your solution was my first idea. Then I spent one day solving tasks with bit operators. I saw the  idea to use n & (n-1) in another task and puzzle was done! I also google one more solution.  
x = n& -n !!!

**Review by belovmd — APPROVED** [review]
10

**Pull request closed without merging**